### PR TITLE
GCS Transfer python CLI integration

### PR DESCRIPTION
**Pull request by Jayancv**

### CODE REVIEW

**Comment by Jayancv** [comment]
@DImuthuUpe Service account creation steps added.

**Comment by DImuthuUpe** [comment]
@Jayancv thanks for the PR and it looks great. Before going further into the implementation, we might have to fix some minor formatting issues. I think your IDE auto-optimizes the code and you might have to update the schema or manually apply the right formatting. (I prefer the latter as it improves your coding discipline). I will point to few examples and please apply them to the rest of the code

File: python-cli/mft_cli/mft_cli/storage/gcs.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+import base64
+import google.auth
+import googleapiclient.discovery
+
+gcs_key_path = '.mft/keys/gcs/service_account_key.json'
+
+
+def handle_add_storage():
+    session_token = ""
+    gcs_regions = ["us-central", "us-east1", "us-east4", "us-east5", "us-south1",
```
[the remark this answers is not in the document]
Comment (DImuthuUpe) [reply python-cli/mft_cli/mft_cli/storage/gcs.py:37]: @Jayancv

**Comment by DImuthuUpe** [comment]
@Jayancv I have added some comments. Also, there is a merge conflict for some reason. Can you check?

**Review by DImuthuUpe — CHANGES_REQUESTED** [review]

File: transport/gcp-transport/src/main/java/org/apache/airavata/mft/transport/gcp/GCSMetadataCollector.java [hunk 2]
```diff
@@ -19,131 +19,224 @@
...
+public class GCSMetadataCollector implements MetadataCollector
+{
 
     boolean initialized = false;
 
     private GCSStorage gcsStorage;
     private GCSSecret gcsSecret;
 
     @Override
-    public void init(StorageWrapper storage, SecretWrapper secret) {
```
[the remark this answers is not in the document]
Comment (Jayancv) [reply transport/gcp-transport/src/main/java/org/apache/airavata/mft/transport/gcp/GCSMetadataCollector.java]: Hope the current version is okay.

File: python-cli/mft_cli/mft_cli/storage/gcs.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+    else:  # Loading credentials from the gcloud cli config file
+        service_account = None
+        client_email = None
+        default_service_account_name = 'mft-gcs-serviceaccount'
+
+        # Find the active config for Google cloud ADC
+        active_config_path = os.path.join(os.path.expanduser('~'), '.config/gcloud/active_config')
+        with open(active_config_path) as f:
+            active_config = f.readline()
+        print('Active config : ' + active_config)
```
[the remark this answers is not in the document]
Comment (DImuthuUpe) [reply python-cli/mft_cli/mft_cli/storage/gcs.py:69]: You can remove this if it disrupts the usability

**Review by DImuthuUpe — CHANGES_REQUESTED** [review]
All looks great @Jayancv. Thanks. Please fix the merge conflict

**Pull request merged**